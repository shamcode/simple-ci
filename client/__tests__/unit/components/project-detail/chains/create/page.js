import { createDI } from 'sham-ui';
import Page from '../../../../../../src/components/project-detail/chains/create/page.sfc';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/lib/href-to';
import renderer from 'sham-ui-test-helpers';

const flushPromises = () => new Promise( resolve => setImmediate( resolve ) );

it( 'renders correctly', () => {
    const DI = createDI();
    const storeMock = jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' )
    } );

    const meta = renderer( Page, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( storeMock.mock.calls[ 0 ][ 0 ] ).toBe( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'create fail', async() => {
    expect.assertions( 2 );

    const DI = createDI();
    const createProjectChain = jest.fn();
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) ),
        createProjectChain: createProjectChain
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValue( '/' )
    } );

    const meta = renderer( Page, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    const { ctx: { container } } = meta;
    await flushPromises();

    const formData = {
        name: 'Test name'
    };
    container.querySelector( '[name="name"]' ).value = formData.name;

    createProjectChain.mockReturnValueOnce( Promise.reject() );
    container.querySelector( '[type="submit"]' ).click();
    expect( createProjectChain.mock.calls ).toHaveLength( 1 );

    await flushPromises();
    expect( meta.toJSON() ).toMatchSnapshot();
} );
