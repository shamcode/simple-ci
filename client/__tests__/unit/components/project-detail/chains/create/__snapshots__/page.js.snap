// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create fail 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "createProjectChain": [Function],
    "dataLoaded": true,
    "dataSaving": false,
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [
      "Create project fail!",
    ],
    "project": Object {
      "cwd": "/tmp/",
      "id": 1,
      "name": "Test",
    },
  },
  "html": "
 <ul class=\\"breadcrumb\\">
   <li>
     <a href=\\"/\\"> Projects </a>
   </li>
   <li>
     <a href=\\"/\\">Test</a>
   </li>
   <!--0-->
   <!--0-->
   <li>Create project chain</li>
 </ul>
 <!--0-->
 <!--0-->
 <ul class=\\"errors-panel\\">
   <li>Create project fail!</li>
 </ul>
 <!--0-->
 <!--0-->
 <div class=\\"project-chain-form\\">
   <form>
     <div class=\\"group\\">
       <input name=\\"name\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
       <label>Name</label>
     </div>
     <div class=\\"group\\">
       <textarea wrap=\\"off\\" style=\\"display: none\\" name=\\"command\\"></textarea>
       <div class=\\"code-flask\\">
         <div class=\\"codeflask codeflask--has-line-numbers\\">
           <textarea class=\\"codeflask__textarea codeflask__flatten\\" spellcheck=\\"false\\" autocapitalize=\\"off\\" autocomplete=\\"off\\" autocorrect=\\"off\\"></textarea><pre class=\\"codeflask__pre codeflask__flatten  language-html\\"><code class=\\"codeflask__code  language-html\\"></code></pre>
           <div class=\\"codeflask__lines\\"><span class=\\"codeflask__lines__line\\">1</span></div>
         </div>
       </div>
       <!--0--><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
       <label>Command</label>
     </div>
     <div class=\\"group\\">
       <button type=\\"submit\\"><span>Create</span></button>
     </div>
   </form>
 </div>
 <!--1-->
 <!--0-->
 <!--1-->
 <!--0-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "createProjectChain": [Function],
    "dataLoaded": false,
    "dataSaving": false,
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {},
  },
  "html": "
 <ul class=\\"breadcrumb\\">
   <li>
     <a href=\\"/\\"> Projects </a>
   </li>
   <!--0-->
   <li></li>
 </ul>
 <!--0-->
 <!--0-->
 <!--0-->
 <!--0-->
 <!--1-->
 <!--0-->
",
}
`;
